.pre-commit-config.yaml

```diff
@@ -1,33 +1,77 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+
+minimum_pre_commit_version: '2.8.2'
+
+default_language_version:
+  python: python3
+
+default_stages: [commit]
+
+
 repos:
+  # General file checks and fixers
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v3.4.0
+    rev: v4.1.0
     hooks:
       - id: mixed-line-ending
-        name: Normalize mixed line endings
+        name: "Normalize mixed line endings"
         args: [--fix=lf]
+      - id: fix-byte-order-marker
+        name: "Remove Unicode BOM"
+      - id: file-contents-sorter
+        name: "Sort codespell ignore list"
+        files: '.codespell/ignore-words.txt'
+
+      - id: check-case-conflict
+        name: "Check for case conflicts"
+      - id: check-merge-conflict
+        name: "Check for merge conflict markers"
+      - id: check-executables-have-shebangs
+        name: "Check that executables have shebangs"
+      - id: check-shebang-scripts-are-executable
+        name: "Check that shebangs are executable"
+
+      - id: check-vcs-permalinks
+        name: "Check that VCS links are permalinks"
```

Comment:
Does this enforce changes like this?

```diff
-https://github.com/markshannon/cpython/blob/vectorcall-minimal/Objects/classobject.c#L53
+https://github.com/markshannon/cpython/blob/815cc1a30d85cdf2e3d77d21224db7055a1f07cb/Objects/classobject.c#L53
```

I understand the utility when linking to a line number (`#L53`), to make sure the link still points to the correct line even if the file changes.

Does the check only trigger for those linking to line numbers, or does it allow things like `https://github.com/markshannon/cpython/blob/vectorcall-minimal/Objects/classobject.c`?

Reply:
> Does this enforce changes like this?

Yup

> Does the check only trigger for those linking to line numbers, or does it allow things like https://github.com/markshannon/cpython/blob/vectorcall-minimal/Objects/classobject.c?

I'd personally find it useful if it did, since multiple times yesterday I ran into non-permalinks to GitHub files that had disappeared completely, and I had to manually rewind time to find them. However, [checking at the source](https://github.com/pre-commit/pre-commit-hooks/blob/c19f885057a6584538f06463fa4bca6c6cb79fc4/pre_commit_hooks/check_vcs_permalinks.py#L10) and testing it to verify, it indeed only enforces this on links to specific lines.